enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,319 @@
+---
+title: azure_user_defined_tags
+authors:
+  - "@bhb"
+reviewers:
+  - "@patrickdillon" ## reviewer for installer component
+  - "@JoelSpeed" ## reviewer for api and machine-api-provider-azure components
+  - "@dmage" ## reviewer for cluster-image-registry-operator component
+  - "@Miciah" ## reviewer for cluster-ingress-operator component
+  - "@akhil-rane" ## reviewer for cloud-credential-operator component
+  - TBD ## reviewer for cloud-network-config-controller component
+approvers:
+  - "@jerpeter1" ## approver for CFE
+api-approvers:
+  - "@JoelSpeed" ## approver for api component
+creation-date: 2022-07-12
+last-updated: 2022-07-12
+tracking-link:
+  - https://issues.redhat.com/browse/OCPPLAN-8155
+  - https://issues.redhat.com/browse/CORS-2249
+see-also:
+  - "enhancements/api-review/custom-tags-aws.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Apply user defined tags to all Azure resources created by OpenShift
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of Openshift to 
+have the ability to apply user defined tags to those resources created by Openshift 
+in Azure.
+
+## Motivation
+
+Motivations include but are not limited to:
+
+- Allow admin, compliance, and security teams to keep track of assets and objects 
+  created by OpenShift in Azure.
+
+### User Stories
+
+- As an openshift administrator, I want to have tags added to all resources created 
+  in Azure by Openshift.
+- As an openshift administrator, I want to restrict access granted to Openshift 
+  specific account.
+
+### Goals
+
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift 
+  can configure allowed number of user-defined tags in the Openshift installer generated
+  install config, which is referred and applied by the installer and the in-cluster operators
+  on the the Azure resources during cluster creation.
+- Tags must be applied at creation time, in an atomic operation. It isn't acceptable 
+  to create an object and to apply tags post cluster creation.
+
+### Non-Goals
+
+- Management(update/delete) of resource tags post creation of cluster is out of scope.
```

Comment:
last i knew we'd killed those efforts, are they back?

Reply:
at the time we dropped them, we surveyed the customers who wanted AWS tagging in general, and they made it clear all they needed was the ability to tag resources at creation time.

so we dropped it:
1) because no one needed it (so why bother w/ the complexity/effort?)
2) updating tags gets into lots of fun questions about how to reconcile conflicts w// manual edits, what to do about deletions, what to do if you exceed the tag count limits, etc.

That said, i think i saw something recently that AWS tag updating did ultimately land, i don't know/remember what brought it back from the dead.